synapseclient/models/mixins/table_operator.py

```diff
@@ -1898,174 +1923,220 @@ async def main():
+                        if cell_value != getattr(row, column):
+                            if (
+                                isinstance(cell_value, list) and len(cell_value) > 0
+                            ) or not isna(cell_value):
+                                partial_change_values[
+                                    column_id
+                                ] = _convert_pandas_row_to_python_types(
+                                    cell=cell_value, column_type=column_type
+                                )
+                            else:
+                                partial_change_values[column_id] = None
+
+                    if partial_change_values != {}:
+                        total_row_count_updated += 1
+                        partial_change = PartialRow(
+                            row_id=row.ROW_ID,
+                            etag=row_etag,
+                            values=[
+                                {
+                                    "key": partial_change_key,
+                                    "value": partial_change_value,
+                                }
+                                for partial_change_key, partial_change_value in partial_change_values.items()
+                            ],
+                        )
+                        rows_to_update.append(partial_change)
+                        indexs_of_original_df_with_changes.append(matching_row.index[0])
+                progress_bar.update(len(individual_chunk))
+            progress_bar.close()
 
         rows_to_insert_df = values.loc[
             ~values.index.isin(indexs_of_original_df_with_changes)
         ]
 
-        client = Synapse.get_client(synapse_client=synapse_client)
         client.logger.info(
-            f"[{self.id}:{self.name}]: Found {len(rows_to_update_df)}"
+            f"[{self.id}:{self.name}]: Found {total_row_count_updated}"
             f" rows to update and {len(rows_to_insert_df)} rows to insert"
         )
 
-        if client.logger.isEnabledFor(logging.DEBUG):
-            client.logger.debug(
-                f"[{self.id}:{self.name}]: Rows to update: {rows_to_update_df}"
-            )
-            client.logger.debug(
-                f"[{self.id}:{self.name}]: Rows to insert: {rows_to_insert_df}"
-            )
+        if not dry_run and rows_to_update:
+            chunked_rows_to_update = []
+            chunk_size_limit = 1.5 * 1024 * 1024  # 1.5MB
+            current_chunk_size = 0
+            chunk = []
+            for row in rows_to_update:
+                row_size = sys.getsizeof(row.to_synapse_request())
+                if current_chunk_size + row_size > chunk_size_limit:
+                    chunked_rows_to_update.append(chunk)
+                    chunk = []
+                    current_chunk_size = 0
+                chunk.append(row)
+                current_chunk_size += row_size
+
+            if chunk:
+                chunked_rows_to_update.append(chunk)
```

Comment:
This is some of the largest changes here. This interface that we use to submit a PartialRow change is what limits requests to 2MB or less:

https://rest-docs.synapse.org/rest/POST/entity/id/table/transaction/async/start.html

In this case I am using 1.5MB to allow for some overhead room.